js/kucoin.js

```diff
@@ -756,10 +757,121 @@ module.exports = class kucoin extends Exchange {
         const request = {
             'currency': currency['id'],
         };
-        const response = await this.privateGetWithdrawalsQuotas (this.extend (request, params));
-        const data = response['data'];
+        const response = await this.publicGetCurrenciesCurrency (this.extend (request, params));
+        //
+        //     v1
+        //     {
+        //         "code":"200000",
+        //         "data":{
+        //             "currency":"USDT",
+        //             "name":"USDT",
+        //             "fullName":"Tether",
+        //             "precision":8,
+        //             "confirms":12,
+        //             "contractAddress":"",
+        //             "withdrawalMinSize":"50",
+        //             "withdrawalMinFee":"25",
+        //             "isWithdrawEnabled":true,
+        //             "isDepositEnabled":true,
+        //             "isMarginEnabled":true,
+        //             "isDebitEnabled":true
+        //         }
+        //     }
+        //
+        //     v2
+        //     {
+        //         "code":"200000",
+        //         "data":{
+        //             "currency":"USDT",
+        //             "name":"USDT",
+        //             "fullName":"Tether",
+        //             "precision":8,
+        //             "confirms":null,
+        //             "contractAddress":null,
+        //             "isMarginEnabled":true,
+        //             "isDebitEnabled":true,
+        //             "chains":[
+        //                 {
+        //                     "chainName":"SOL",
+        //                     "withdrawalMinSize":"2",
+        //                     "withdrawalMinFee":"1",
+        //                     "isWithdrawEnabled":false,
+        //                     "isDepositEnabled":true,
+        //                     "confirms":32,
+        //                     "contractAddress":""
+        //                 }
+        //             ]
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data');
+        const versions = this.safeValue (this.options, 'versions', {});
+        const apiVersions = this.safeValue (versions, 'public', {});
+        const methodVersions = this.safeValue (apiVersions, 'GET', {});
+        const defaultVersion = this.safeString (methodVersions, 'currencies/{currency}', this.options['version']);
+        const version = this.safeString (params, 'version', defaultVersion);
+        const withdrawFees = {};
+        if (version === 'v1') {
+            withdrawFees[code] = this.safeNumber (data, 'withdrawalMinFee');
```

Comment:
Question, for V1 can't we infer the network?

Reply:
I think we can use a custom map in options for that, but that might looks weird?

Reply:
@sc0Vu yeah probably, but at least we should check if the user provided it no? In that case we can add it to the response

![image](https://user-images.githubusercontent.com/43336371/180964161-59ccb0a4-8508-40ae-ab4d-0d35c7c8e763.png)